### PR TITLE
scripts: Add data collection script

### LINKED ISSUE #80: Add data collection script
Add the data collection script [*] that generates output users can paste into a github issue when reporting issues.

---

[*] - https://github.com/clearcontainers/runtime/blob/master/data/collect-data.sh.in

### PR DESCRIPTION
**Pull request by jodh-intel**
Add a data collection script that can be run by users and its output
pasted directly into a github issue.

The script is designed to make diagnosing issues as easy as possible
and its output provides a summary of a Kata Containers system
including:

- Versions of all components.
- Details of container managers.
- Errors found in the system journal.

Fixes #80.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
Ping @bergwolf, @grahamwhaley, @sboeuf.

File: data/kata-collect-data.sh.in [hunk 1]
```diff
@@ -0,0 +1,570 @@
+#!/bin/bash
+#
+# Copyright (c) 2017-2018 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (grahamwhaley) [inline data/kata-collect-data.sh.in]: SPDX header style?

File: data/kata-collect-data.sh.in [hunk 2]
```diff
@@ -0,0 +1,570 @@
...
+}
+
+# Returns: Full path to the image file.
+get_image_file()
+{
+	local cmd="@PROJECT_TYPE@-env"
+	local cmdline="$runtime $cmd"
+
+	image=$(eval "$cmdline" 2>/dev/null |\
+		grep -A 1 '\[Image\]' |\
```
Comment (grahamwhaley) [inline data/kata-collect-data.sh.in]: Maybe `'^\[Image\]'` to be sure we pick up that top level section?
The `grep -A 1` is obviously a bit tied to the exact output of the `xxx-env` command - probably not an issue, but could fall over one day.

I did something v.similar a couple of days ago with some awk like:
```
	RUNTIME_PATH=$(awk '/^\[Runtime\]$/ {foundit=1} /^    Path =/ { if (foundit==1) {print $3; foundit=0} } ' <<< "$toml" | sed 's/"//g')
```

just for reference. Also not perfect I think.

File: data/kata-collect-data.sh.in (hunk 1 above)
Comment (jodh-intel) [reply data/kata-collect-data.sh.in]: Fixed.

File: data/kata-collect-data.sh.in (hunk 2 above)
Comment (jodh-intel) [reply data/kata-collect-data.sh.in]: I've added the anchor and I agree about the fagility issue, but until we start producing output in a superior machine readable format (JSON, but ideally YAML), I think we're going to continue to have this risk.

Did you mention you were going to add JSON support to `kata-env`... ? :smile:

File: data/kata-collect-data.sh.in [hunk 3]
```diff
@@ -0,0 +1,561 @@
...
+
+show_package_versions()
+{
+	heading "Packages"
+
+	local pattern="("
+	local project
+
+	# CC 2.x runtime. This shouldn't be installed but let's check anyway
+	pattern+="cc-oci-runtime"
```
Comment (devimc) [inline data/kata-collect-data.sh.in]: here we are considering to cc-oci-runtime (2.x), I wonder if cc-runtime (3.0) should be consider too

File: data/kata-collect-data.sh.in [hunk 4]
```diff
@@ -0,0 +1,561 @@
...
+	# core components
+	for project in @PROJECT_TYPE@
+	do
+		pattern+="|${project}-proxy"
+		pattern+="|${project}-runtime"
+		pattern+="|${project}-shim"
+	done
+
+	# assets
+	pattern+="|clear-containers-image"
```
Comment (devimc) [inline data/kata-collect-data.sh.in]: might be `${project}-containers-image` ?

File: data/kata-collect-data.sh.in [hunk 5]
```diff
@@ -0,0 +1,561 @@
...
+	pattern+="|linux-container"
+
+	# optimised hypervisor
+	pattern+="|qemu-lite"
+
+	# default distro hypervisor
+	pattern+="|qemu-system-x86"
```
Comment (devimc) [inline data/kata-collect-data.sh.in]: would be a nice to have `aarch64` and `arm`

**Pull request merged**